src/snowflake/snowpark/mock/_functions.py

```diff
@@ -277,32 +278,40 @@ def mock_to_date(
         [x] For NULL input, the output is NULL.
 
         [ ] For all other values, a conversion error is generated.
+
+    # TODO: switch the implementatio to use `try_convert`
     """
     res = []
-    auto_detect = bool(not fmt)
-
-    date_format, _, _ = convert_snowflake_datetime_format(
-        fmt, default_format="%Y-%m-%d"
+    default_date_format, _, _ = convert_snowflake_datetime_format(
+        None, default_format="%Y-%m-%d"
     )
 
-    for data in column:
+    for idx, data in enumerate(column):
```

Comment:
I think there's a better way to do this that is more in line with pandas style. See here:
https://github.com/snowflakedb/snowpark-python/compare/fix-local-testing-to-date-implementation...fix-local-testing-to-date-implementation-jrose